### ISSUE TITLE
Unassigned instantiated classes should trigger a pointless-statement or a new message to be determined

### ISSUE THREAD

**Issue by vapier** [PyCQA/pylint#3110]
### Current status

pylint will issue `pointless-statement` when a class (e.g. Exception) is used.  this is great:
```sh
$ cat test.py
Exception
$ pylint3 test.py
test.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
```

but if the exception is instantiated, it's not shown:
```sh
$ cat test.py
Exception()
$ pylint3 test.py
<nothing>
```

### Proposal

while instantiating random classes can have side-effects, exceptions rarely (if ever) have such behavior.  it's so uncommon that, imo, it should trigger a warning by default.  we ran into issues where people forgot to `raise` the exception (because the naming made it sound like it would throw an error), so code that was supposed to fail ended up not.

so how about:
```sh
$ cat test.py
Exception()
$ pylint3 test.py
test.py:1:0: W0104: Instantiated object seems to have no effect (pointless-exception-object)
```

feel free to bikeshed the exact naming/phrasing.

### Version info

```sh
$ pylint3 --version
pylint3 2.2.2
astroid 2.1.0
Python 3.6.8 (default, Jan  3 2019, 03:42:36) 
[GCC 8.2.0]
```

**Comment by vapier** [comment]
maybe something like:
```python
    def visit_expr(self, node):
        if not isinstance(node.value, astroid.Call):
            return
        func = node.value.func
        try:
            cls = next(func.infer())
        except astroid.InferenceError:
            return
        if not isinstance(cls, astroid.ClassDef):
            return
        if any(x for x in cls.ancestors() if x.name == 'BaseException'):
            self.add_message('...', node=node, line=node.fromlineno)
```

**Comment by PCManticore** [comment]
This sounds good to me, but I think we should do it for the general case not just for exceptions. Instantiating a class and not doing anything with it sounds like a smelling to me (note that might have a purpose, e.g. testing that the instantiation raises an exception)

**Comment by PCManticore** [comment]
@vapier From my understanding, this new check will handle the following case:
```
Class()
```

while ```Class().run()``` and `c = Class(); c.run()` will not be caught.

The former looks more of an application bug than a code smell, as that instantiation might miss an additional method call or a variable assignment.

**Comment by Pierre-Sassoulas** [comment]
This is still relevant as 
```python
from pathlib import Path

Exception()
Path()
```
does not raise any message.

**Comment by jayaddison** [comment]
This issue is now covered by a check in `pylint` v2.16:

```
$ cat test.py 
Exception()
$ pylint test.py
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: W0133: Exception statement has no effect (pointless-exception-statement)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for the help triaging @jayaddison :) Indeed one of the highlights of 2.16 for me !
